api/pkg/transformer/symbol/time.go

```diff
@@ -0,0 +1,241 @@
+		return function.DayOfWeek(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// IsWeekend check wheter given timestamps falls in weekend, given timestamp and timezone
+// timestamp ana timezone can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) IsWeekend(timestamp, timezone interface{}) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.IsWeekend(ts, tz)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// FormatTimestamp will convert timestamp into formatted date string
+// timestamp can be:
+// - Json path string
+// - Slice / gota.Series
+// - int64 value
+func (sr Registry) FormatTimestamp(timestamp, timezone interface{}, format string) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return function.FormatTimestamp(ts, tz, format)
+	}
+	return sr.processTimestampFunction(timestamp, timezone, timeFn)
+}
+
+// ParseTimeStamp convert timestamp value into time
+func (sr Registry) ParseTimestamp(timestamp interface{}) interface{} {
+	timeFn := func(ts int64, tz *time.Location) interface{} {
+		return time.Unix(ts, 0).In(tz)
+	}
+	// empty timezone mean using UTC
+	return sr.processTimestampFunction(timestamp, "", timeFn)
+}
+
+func (sr Registry) processTimestampFunction(timestamps, timezone interface{}, timeTransformerFn func(timestamp int64, tz *time.Location) interface{}) interface{} {
+	ts, err := sr.evalArg(timestamps)
+	if err != nil {
+		panic(err)
+	}
+
+	tz, err := sr.evalArg(timezone)
+	if err != nil {
+		panic(err)
+	}
+
+	timestampVals := reflect.ValueOf(ts)
+	timezoneVals := reflect.ValueOf(tz)
+
+	switch timestampVals.Kind() {
+	case reflect.Slice:
+		if timestampVals.Len() == 0 {
+			panic("empty array of timestamp provided")
+		}
+
+		location := &time.Location{}
+		if timezoneVals.Kind() == reflect.Slice {
+			if timezoneVals.Len() == 0 {
+				panic("empty array of timezone provided")
+			}
+
+			if timestampVals.Len() != timezoneVals.Len() {
+				panic("both timestamp and timezone arrays must have the same length")
+			}
+		} else {
+			timeLocation, err := LoadLocationCached(fmt.Sprintf("%s", tz))
+			if err != nil {
+				panic(err)
+			}
+			location = timeLocation
+		}
+
+		var values []interface{}
+		for idx := 0; idx < timestampVals.Len(); idx++ {
+			val := timestampVals.Index(idx)
+			tsInt64, err := converter.ToInt64(val.Interface())
+			if err != nil {
+				panic(err)
+			}
+
+			if timezoneVals.Kind() == reflect.Slice {
+				timeLocation, err := LoadLocationCached(fmt.Sprint(timezoneVals.Index(idx)))
+				if err != nil {
+					panic(err)
+				}
+				location = timeLocation
+			}
+
+			values = append(values, timeTransformerFn(tsInt64, location))
+		}
+		return values
+	default:
+		tsInt64, err := converter.ToInt64(ts)
+		if err != nil {
+			panic(err)
+		}
+
+		timeLocation, err := LoadLocationCached(fmt.Sprintf("%s", tz))
```

Comment:
what happen if `tz` is:
* slice with single value
* slice with multiple values

Reply:
it will return error, hence panic because the location is not found

Reply:
do you think we need to validate the timezone data type here? we expect timezone to be single value to match timestamp

Reply:
I think it's better to be defensive and check the type as well